src/components/content/Link.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { request } from 'graphql-request'
+import { idQuery } from '../../fetcher/query'
+import { endpoint } from '../../fetcher/serlo-api'
+import StyledA from '../tags/StyledA'
+import ExternalLink from './ExternalLink'
+
+const nowrap = comp => comp
+
+export default function Link({
+  element,
+  attributes = {},
+  children = null,
+  wrapExtInd = nowrap
+}) {
+  const [prettyHref, setPrettyHref] = React.useState('')
+
+  const isExternal = element.href.indexOf('//') > -1
+  const isId = /^\/[\d]+$/.test(element.href)
+
+  React.useEffect(() => {
+    if (!isId) return
+
+    request(endpoint, idQuery(element.href.substring(1))).then(res => {
+      if (res && res.uuid && res.uuid.alias) setPrettyHref(res.uuid.alias)
+    })
+  }, [])
+
+  if (!element.href)
+    return <React.Fragment {...attributes}>{children}</React.Fragment>
+
+  return (
+    <StyledA href={prettyHref ? prettyHref : element.href} {...attributes}>
```

Comment:
Doesn't this still use the non-pretty href on server? (which is IMHO the important part were we want the pretty href because Google)

Reply:
totally right, was only focusing on UX

Reply:
will try to make it work on server

Reply:
Just to make sure: would it be appropriate to use `getServerSideProps` in this case?

Reply:
If you get the server side props down to the Link component somehow, sure. We probably need to handle that universally in the API fetch logic we have currently in place, e.g.
- Fetch basic data by alias or id (that is what we currently do)
- Go over that and make a list of links we need to prettify
- Fetch pretty alias for that links
- Pass that down to the page (and to the links via context)

This way, Vercel can still cache the API request and the server-side rendered stuff contains the pretty links.

Reply:
sorry, this is getting more complex then anticipated (well that never happened before :).
Maybe one last question before we move this to a later stage: Are you proposing we basically go through the returned `data.value.children` and collect all the `a` and build a new query to get their pretty links?

Reply:
Yep. If you see another way, I'm all ears. The main issue we need to work around is: we can't do `getServerSideProps` on component level (only on page level). This means: the page itself must fetch the data itself (since we want them during server-side rendering) and pass them down somehow via context.

Reply:
links as async components? could that work? otherwise I'll try your suggestion.

Reply:
scratch that, I thought privacy was loaded this way, but it's front end only as well.